src/Controller/LoginController.php

```diff
@@ -0,0 +1,34 @@
+<?php declare(strict_types=1);
+
+namespace Pdsinterop\Solid\Controller;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+class LoginController extends AbstractController
+{
+    final public function __invoke(ServerRequestInterface $request, array $args): ResponseInterface
+    {
+        $postBody = $request->getParsedBody();
+        $response = $this->getResponse();
+        // var_dump($_SESSION);
+        if (isset($_SESSION["userid"])) {
+          $user = $_SESSION["userid"];
+          $response->getBody()->write("<h1>Already logged in as $user</h1>");
+        } else if ($postBody["user"] == $_ENV["USER"] && $postBody["password"] == $_ENV["PASSWORD"]) {
+          $user = $postBody["user"];
+          $response->getBody()->write("<h1>Welcome $user</h1>\n");
+          $_SESSION["userid"] =  $user;
+          print_r("session started\n");
+          var_dump($_SESSION);
+        } else {
+          // var_dump($postBody);
+          print_r("cookie:\n");
```

Comment:
`print_r` is more commonly used to output variables, not string, `echo` is prefered for plain strings.

Pro-tip: If you want the output from `var_dump` to be present in the response body, you can also use:

```php
           $response->getBody()->write(var_export($_SESSION, true));  
```

The `true` means "return instead of output".